src/Http/Auth/RequestAuthenticatorMiddleware.php

```diff
@@ -91,7 +91,7 @@ private function authenticate(ServerRequestInterface $request): void
         // Requests that use a token that they got directly from Auth0 do not require an API key.
         // The difference can be checked by checking if the token has a client id, which is always missing in tokens
         // from the JWT providers.
-        if ($this->token->getClientId() === null) {
+        if ($this->token->getClientId() === null || $this->token->getType() === JsonWebToken::UIT_ID_V2_JWT_PROVIDER_TOKEN) {
```

Comment:
I like how you made the check more readable here by checking the token type. Maybe you can also do that for the first check (empty client id) by checking for V1 JWT provider type instead?